Open/pages/referee/_id.vue

```diff
@@ -1033,18 +1071,82 @@ export default class Referee extends Mixins(CentrifugeMixin) {
+        let currentScrollHeight = 0;
+        if (messageContainer) // Null in the case of mounted and no mp property
+            currentScrollHeight = messageContainer.scrollHeight;
+        const { data: messagesData } = await this.$axios.get<{ messages: MatchupMessageBasic[] }>(`/api/referee/matchups/${this.tournament?.ID}/${this.matchup?.ID}/messages${this.messages[0]?.ID ? `?before=${this.messages[0]?.ID}` : ""}`);
+        if (!messagesData.success) {
+            alert("Failed to fetch messages. Check console for more information.");
+            console.error(messagesData.error);
+            this.loadMoreMessages = false;
+        } else {
+            const newMessages = messagesData.messages.map(message => ({
+                ...message,
+                timestamp: new Date(message.timestamp),
+            })).reverse();
+
+            // If messages were obtained from centrifuge history and this is the first time loading messages, there may be duplicates (based on content, and timestamp, see https://github.com/Corsace/corsace/blob/master/BanchoBot/functions/tournaments/matchup/runMatchup.ts#L234) with different IDs, so this removes the duplicates obtained from centrifuge history
+            if (this.messages.length > 0 && newMessages.length > 0 && this.messages[0].ID === 0) {
+                // Centrifuge history messages will have an ID based on the length of the messages array, so this can be used to loop and break, as messages from the DB will have a completely different ID
+                for (let i = 0; i < this.messages.length; i++) {
+                    if (this.messages[i].ID !== i) 
+                        break;
+
+                    if (this.messages[i].content === newMessages[newMessages.length - 1].content && this.messages[i].timestamp.getTime() === newMessages[newMessages.length - 1].timestamp.getTime())
+                        this.messages.shift();
+                }
+            }
+
+            this.messages = [
+                ...newMessages,
+                ...this.messages,
+            ];
+
+            this.loadMoreMessages = messagesData.messages.length === 50;
+            
+            await this.$nextTick();
+            messageContainer = document.getElementById("messageContainer"); // In case it was null before and now it's not
+            if (messageContainer && messageContainer.scrollHeight === messageContainer.clientHeight && this.loadMoreMessages) {
+                await this.loadMessages(toBottom);
+                return;
+            }
+
+            if (toBottom)
+                this.scrollToBottom();
+            else if (messageContainer) {
+                const newScrollHeight = messageContainer.scrollHeight;
+                messageContainer.scrollTo({
+                    top: messageContainer.scrollTop + newScrollHeight - currentScrollHeight,
+                    behavior: "auto",
+                });
+            }
+        }
+        this.loadingMessages = false;
+    }
```

Comment:
this should be wrapped in a try-finally so that a single error doesn't soft-lock the chat, eg:
```
this.loadingMessages = true;
try {
  // the entire function
} finally {
  this.loadingMessages = false;
}
```

also, this isnt critical and the invoked method should probably continue running if this fails. so add some error handling to this, either in the function with a catch in between try and finally, or on the call

Reply:
does
```
this.loadingMessages = true;
try {
  // the entire function
} catch (e) {
  console.error(e);
}
this.loadingMessages = false;
```
work or does that change the flow somehow where it wont run the final line

Reply:
oh nvm i forgot about the return